asteroid/filterbanks/simple_adaptive.py

```diff
@@ -0,0 +1,111 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class AdaptiveEncoder1D(nn.Module):
+    """
+        A 1D convolutional block that transforms signal in wave form into higher
+        dimension.
+
+        Args:
+            input shape: [batch, 1, n_samples]
+            output shape: [batch, freq_res, n_samples//sample_res]
+            freq_res: number of output frequencies for the encoding convolution
+            sample_res: int, length of the encoding filter
+    """
+
+    def __init__(self, freq_res, sample_res):
+        super().__init__()
+        self.conv = nn.Conv1d(1,
+                              freq_res,
+                              sample_res,
+                              stride=sample_res // 2,
+                              padding=sample_res // 2)
+
+    def forward(self, s):
+        return F.relu(self.conv(s))
+
+
+class AdaptiveDecoder1D(nn.Module):
+    """ A 1D deconvolutional block that transforms encoded representation
+    into wave form.
+    input shape: [batch, freq_res, sample_res]
+    output shape: [batch, 1, sample_res*n_samples]
+    freq_res: number of output frequencies for the encoding convolution
+    sample_res: length of the encoding filter
+    """
+
+    def __init__(self, freq_res, sample_res, n_sources):
+        super().__init__()
+        self.deconv = nn.ConvTranspose1d(n_sources * freq_res,
+                                         n_sources,
+                                         sample_res,
+                                         padding=sample_res // 2,
+                                         stride=sample_res // 2,
+                                         groups=n_sources,
+                                         output_padding=(sample_res // 2) - 1)
```

Comment:
This makes one decoder per source, I don't think this is what we want. 
Please have a look at `FreeFB`, we could add `padding` and `output_padding` to `Encoder` and `Decoder` to cover your intended usage, it would be a good addition :smiley: